src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const arr = [...this];
+
+    for (let i = 0; i < arr.length; i++) {
+      if (arr[i] === null || arr[i] === undefined) {
+        arr[i] = '';
```

Comment:
Here gonna be better if you add at least a separator to the final string, for example, if you have [undefined, 123, null, 'abc'] result will be like ', 123, , abc'. It indicates, that there was a value in the array. And it is how method .join really work